builtin/rebase.c

```diff
@@ -134,6 +134,8 @@ static int run_specific_rebase(struct rebase_options *opts)
 			"verbose=t" : "verbose");
 	argv_array_pushf(&env, opts->flags & REBASE_DIFFSTAT ?
 			 "diffstat=t" : "diffstat");
+	argv_array_pushf(&env, "GIT_REFLOG_ACTION=%s",
+			opts->type & REBASE_INTERACTIVE);
```

Comment:
That is not going to work. What you probably wanted to do here is to have a [ternary operator](https://en.wikipedia.org/wiki/%3F:), with `opts->type == REBASE_INTERACTIVE` as condition (not `opts->type & REBASE_INTERACTIVE` because `type` is a regular `enum` type, not a bit field), and then imitates what https://github.com/prertik/git/blob/ae91dbb2556d7ea9f4e205fbf7847fa8579c64b4/TODO-rebase.sh#L416-L421 does (but to make it a real ternary operator, you would have to append `? "rebase -i (<action>)" : "rebase"`, which does not work because there is no primitive in C for string concatenation/interpolation).

Besides, the shell script version goes out of its way to respect an already-set `GIT_REFLOG_ACTION` and *not* override it.

So I think you need to do this differently. Something along these lines:

```c
#define GIT_REFLOG_ACTION_ENVIRONMENT "GIT_REFLOG_ACTION"

/* only set the reflog message if it has not been set yet */
if (!getenv(GIT_REFLOG_ACTION_ENVIRONMENT)) {
    struct strbuf buf = STRBUF_INIT;
    strbuf_addf(&buf, "%s=rebase", GIT_REFLOG_ACTION_ENVIRONMENT);
    if ((opts->type == REBASE_INTERACTIVE ||
         opts->type == REBASE_PRESERVE_MERGES) &&
        opts->action_name)
        strbuf_addf(&buf, " -i (%s)", opts->action_name);
    argv_array_pushf(&env, buf.buf);
    strbuf_release(&buf);
}
```

For this to work, you will have to add the `action_name` field to `rebase_options`. Obviously, it needs to have the type `const char *`.

And while you could play games with adding the actions as `OPT_CMDMODE()` in the exact same order as the enum and then pluck the action name from the `long` field, I think it would be a lot more robust and quite a bit easier to review by adding them explicitly in the `switch ()` that you will already [have to add to handle the actions](https://github.com/prertik/git/blob/ae91dbb2556d7ea9f4e205fbf7847fa8579c64b4/TODO-rebase.sh#L429-L471):

```c
switch (action) {
case TODO_CONTINUE:
    options.action_name = "continue";
    ...
    break;
case TODO_SKIP:
    options.action_name = "skip";
    ...
...
```